bolt-sidecar/src/builder/template.rs

```diff
@@ -111,6 +111,18 @@ impl BlockTemplate {
             .fold(0, |acc, sc| acc + sc.message.constraints.len())
     }
 
+    /// Returns the committed gas in the block template.
+    #[inline]
+    pub fn committed_gas(&self) -> U256 {
+        self.signed_constraints_list
+            .iter()
+            .fold(U256::ZERO, |acc, sc| {
+                acc + sc.message.constraints.iter().fold(U256::ZERO, |acc, c| {
+                    acc + max_transaction_cost(&c.transaction)
+                })
+            })
+    }
```

Comment:
committed gas should keep track of the amount of gas units used by all commitments, therefore it should sum the `gas_limit` of all transactions rather than the `max_transaction_cost`